source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -100,16 +104,77 @@ void EnvoyQuicClientConnection::switchConnectionSocket(
   quic::QuicSocketAddress peer_address = envoyIpAddressToQuicSocketAddress(
       connection_socket->connectionInfoProvider().remoteAddress()->ip());
 
-  // The old socket is closed in this call.
+  // The old socket is not closed in this call.
```

Comment:
We shouldn't close the previous socket mainly because there might still be packets in the wild addressing the old address.
This behavior can be seen in Chrome's socket migration as well. https://source.chromium.org/chromium/chromium/src/+/main:net/quic/quic_chromium_client_session.cc;l=3647;drc=99d4c69c9ec60042750e31e02802f0ade763dcd0?q=quic_chromium_client_session&ss=chromium

Reply:
Can you comment that we need to keep the old socket for a while because of incoming packets on the old path?

Reply:
done.